[Lutfi] iP

luffingluffy opened the pull request:
# Biscuit Chat Bot
>"When it is wrong, things are very wrong" -Zhu Chengbo

Biscuit frees your mind of having to remember things you need to do. It's:
- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is: 
1. download it from [here](https://github.com/luffingluffy/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE!**

Features:
- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:

```
public class Duke {
	public static void main(String[] args) {

		init();
		ui.greet();
		reply();
	}
}
```

Commit:
Add Gradle support

luffingluffy committed:
Level-1

luffingluffy committed:
Level-1

luffingluffy committed:
Level-2

luffingluffy committed:
Level-3

luffingluffy committed:
Level-4

luffingluffy committed:
Level-5

luffingluffy committed:
Level-6

luffingluffy committed:
Level-6

luffingluffy committed:
Change taskList Array to Arraylist

luffingluffy committed:
Level-7

luffingluffy committed:
Update .gitignore

luffingluffy committed:
Level-8

luffingluffy committed:
Merge branch-Level-7 with master

luffingluffy committed:
Merge branch-Level-8 with master

luffingluffy committed:
Add TaskList for OOP

luffingluffy committed:
Refactor TaskList implementation

luffingluffy committed:
Add Ui class

luffingluffy committed:
Add storage package

luffingluffy committed:
Add Parser class

luffingluffy committed:
Add TaskTest

luffingluffy committed:
Add JavaDoc

luffingluffy committed:
Refactor to follow coding standard

luffingluffy committed:
Level-9

luffingluffy committed:
Merge branch 'branch-A-JavaDoc'

luffingluffy committed:
Merge branch 'branch-A-CodingStandard'

luffingluffy committed:
Merge branch 'branch-Level-9'

luffingluffy committed:
Merge remote-tracking branch 'origin/add-gradle-support'

chowRuiYan commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,115 @@
+package duke.storage;
+
```
I think you might have missed out on the additional blank line here between your `package`  statement and `import` statements.

chowRuiYan commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+    // instance variables
+    private LocalDateTime deadline;
+
+    // constructors
```
I feel that this comment can be removed since there is a Javadoc for it already. I noticed the same issue in several other places too.

chowRuiYan commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,95 @@
+    }
+
+    /**
+     * Helper method to format the date according to the
```
I think that there isn't a need to indicate that it is a helper method.

Perhaps this would be a better start to your Javadoc. 
```suggestion
     * Formats the date according to the
```

I noticed the same issue at one other place too.

chowRuiYan reviewed (commented):
Looks great! I just added some personal opinions 😄

WuaaAj commented on src/main/java/duke/storage/Storage.java:
```diff
@@ -0,0 +1,115 @@
+    // instance variable
+    private final String filePath;
+
+    // constructors
```
Maybe can remove this line.

WuaaAj commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,107 @@
+     * @param taskList  the list containing the tasks
+     * @throws DukeException if the inputs are not expected
+     */
+    public static void parse(String userInput, TaskList taskList) throws DukeException {
```
The parser function is a bit long. Can seperate the cases into diffrent methods.

WuaaAj commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,96 @@
+    private LocalDateTime date;
+
+    // constructors
+
```
May consider removing the redundant empty lines.

WuaaAj reviewed (commented):
I find your code very readable and easy to understand. Just some minor things can be improved.

luffingluffy committed:
Add A-Gradle and A-Checkstyle

luffingluffy committed:
Add GUI template

luffingluffy committed:
Level-10

luffingluffy committed:
Add assertions

luffingluffy committed:
Improve code readability and quality

This commit improves code readability and quality by grouping output messages together.

In future iterations, we may consider refactoring the commands into its own separate class.

Commit:
Merge pull request #1 from luffingluffy/branch-A-Assertions

Add assertions

luffingluffy committed:
Merge master to branch

Commit:
Merge pull request #2 from luffingluffy/branch-A-CodeQuality

Branch A-CodeQuality

Commit:
Add CI

luffingluffy committed:
Add Snooze function.

There is no snooze function.

Users might want to reschedule the date
for Event/Deadline tasks.

Add a feature for users to reschedule using
snooze {task number} /to {new date using dd:MM:yyyy HH:mm} format.

luffingluffy committed:
Implement CI

luffingluffy committed:
There is no snooze function.

Users might want to reschedule the date
for Event/Deadline tasks.

Add a feature for users to reschedule using
snooze {task number} /to {new date using dd:MM:yyyy HH:mm} format.

luffingluffy committed:
Enhance GUI display

luffingluffy committed:
Refactor TaskList to conform to checkstyle

luffingluffy committed:
Finalise GUI

luffingluffy committed:
Finalise features for release

luffingluffy committed:
Add Ui for release

Commit:
Set theme jekyll-theme-tactile

luffingluffy committed:
Add User Guide

luffingluffy committed:
Merge remote-tracking branch 'origin/master'

luffingluffy committed:
Fix User Guide screenshot

luffingluffy committed:
Minor GUI fix for release

Commit:
Update README.md
